docs/fiddles/windows/manage-windows/window-events/preload.js

```diff
@@ -0,0 +1,9 @@
+const { contextBridge, ipcRenderer } = require('electron')
+
+contextBridge.exposeInMainWorld('electronAPI', {
+  showDemoWindow: () => ipcRenderer.send('show-demo-window'),
+  focusDemoWindow: () => ipcRenderer.send('focus-demo-window'),
+  onWindowFocus: (callback) => ipcRenderer.on('window-focus', callback),
+  onWindowBlur: (callback) => ipcRenderer.on('window-blur', callback),
+  onWindowClose: (callback) => ipcRenderer.on('window-close', callback)
```

Comment:
Best security practice here is to strip the event parameter before calling the callback, e.g.

```js
onWindowFocus: (callback) => ipcRenderer.on('window-focus', () => callback()),
```

Otherwise we leak `event.sender` to the renderer and it can send arbitrary IPC.

Reply:
btw we had this in the existing code https://github.com/electron/electron/blob/d42a94dddeb86d349c050c2ee59b135bd6c8245c/docs/fiddles/ipc/pattern-3/preload.js#L1-L5
I am fixing all the occurences